File: frontend/src/services/catatanService.js
import api from './api'

export default {
  getByProgramKerja(programKerjaId) {
    return api.get(`/catatan/program-kerja/${programKerjaId}`)
  },
  create(programKerjaId, data) {
    return api.post(`/catatan/program-kerja/${programKerjaId}`, data)
  },
  delete(id) {
    return api.delete(`/catatan/${id}`)
  },
}

File: src/main/java/com/ukm/programkerja/service/CatatanService.java
package com.ukm.programkerja.service;

import com.ukm.programkerja.dto.request.CatatanRequest;
import com.ukm.programkerja.dto.response.CatatanResponse;

import java.util.List;

public interface CatatanService {

    /**
     * Menambahkan catatan baru. penulisId diambil dari user yang sedang
     * login (WKIII atau PEMBINA), bukan dari request body.
     */
    CatatanResponse create(Long programKerjaId, Long penulisId, CatatanRequest request);

    List<CatatanResponse> getByProgramKerjaId(Long programKerjaId);

    void delete(Long id);
}
